src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -182,6 +183,23 @@ class ReferralController(
     return referralService.getCancellationReasons()
   }
 
+  @GetMapping("sent-referral/{id}/supplier-assessment")
+  fun getSupplierAssessmentAppointment(
+    @PathVariable id: UUID,
+    authentication: JwtAuthenticationToken,
+  ): SupplierAssessmentDTO {
+    val user = userMapper.fromToken(authentication)
+
+    val sentReferral = referralService.getSentReferralForUser(id, user)
+      ?: throw ResponseStatusException(HttpStatus.NOT_FOUND, "sent referral not found [id=$id]")
+
+    if (sentReferral.supplierAssessment!!.appointments.size == 0) {
```

Comment:
what's the reason this is here?

Reply:
also i don't like the not null assertion on supplierAssessment. we can get a not null reference to the SA like this:

```
val assessment = sentReferral.supplierAssessment ?: throw(not found....)